internal/pkg/manifest/storage.go

```diff
@@ -49,15 +51,15 @@ func (e *EFSVolumeConfiguration) IsEmpty() bool {
 	return e.FileSystemID == nil && e.RootDirectory == nil && e.AuthConfig == nil && e.UID == nil && e.GID == nil
 }
 
-// EFSConfigOrID contains custom unmarshaling logic for the `efs` field in the manifest.
-type EFSConfigOrID struct {
-	Config EFSVolumeConfiguration
-	ID     string
+// EFSConfigOrBool contains custom unmarshaling logic for the `efs` field in the manifest.
+type EFSConfigOrBool struct {
+	Config  EFSVolumeConfiguration
+	Enabled *bool
```

Comment:
Recently with the spot feature we introduced this pattern to handle "advanced" configs:
https://github.com/aws/copilot-cli/blob/2c8bbc122130cb3b6035768d3edc6613a57bb861/internal/pkg/manifest/svc.go#L117-L122

What do you think of refactoring like this:
```suggestion
// EFS holds the configuration for the `efs` field.
type EFS struct {
	Advanced  EFSVolumeConfiguration
	Enabled  *bool
```